impeller/renderer/backend/vulkan/render_pass_builder_vk.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_IMPELLER_RENDERER_BACKEND_VULKAN_RENDER_PASS_BUILDER_VK_H_
+#define FLUTTER_IMPELLER_RENDERER_BACKEND_VULKAN_RENDER_PASS_BUILDER_VK_H_
+
+#include <map>
+#include <optional>
+
+#include "impeller/core/formats.h"
+#include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "impeller/renderer/backend/vulkan/vk.h"
+
+namespace impeller {
+
+class RenderPassBuilderVK {
+ public:
+  RenderPassBuilderVK();
+
+  ~RenderPassBuilderVK();
+
+  RenderPassBuilderVK(const RenderPassBuilderVK&) = delete;
+
+  RenderPassBuilderVK& operator=(const RenderPassBuilderVK&) = delete;
+
+  RenderPassBuilderVK& SetColorAttachment(size_t index,
+                                          PixelFormat format,
+                                          SampleCount sample_count,
+                                          LoadAction load_action,
+                                          StoreAction store_action);
+
+  RenderPassBuilderVK& SetDepthStencilAttachment(PixelFormat format,
+                                                 SampleCount sample_count,
+                                                 LoadAction load_action,
+                                                 StoreAction store_action);
+
+  RenderPassBuilderVK& SetStencilAttachment(PixelFormat format,
+                                            SampleCount sample_count,
+                                            LoadAction load_action,
+                                            StoreAction store_action);
+
+  vk::UniqueRenderPass Build(const vk::Device& device) const;
+
+ private:
+  std::map<size_t, vk::AttachmentDescription> colors_;
+  std::map<size_t, vk::AttachmentDescription> resolves_;
+  std::optional<vk::AttachmentDescription> depth_stencil_;
+};
+
+void InsertBarrierForInputAttachmentRead(const vk::CommandBuffer& buffer,
+                                         const vk::Image& image);
```

Comment:
This method doesn't really seem related to the render_pass builder. Since this is only used in render_pass_vk.cc please move it there

Reply:
Thats not accurate. The access masks and dependency flags must be a subset of the masks and flags specified during subpass self-dependency setup. Those masked are just a few lines above in the implementation. I wanted to keep these two together because the barrier you insert will be closely tied to the self dependency setup, in `render_pass_builder.cc`. I suppose I can make those use a common constant to make that clearer. And also document this either way.

Reply:
I documented and used constants as well adding more context.